src/vehicle.cpp

```diff
@@ -2995,6 +3007,32 @@ void vehicle::center_of_mass(int &x, int &y) const
     y = round(yf);
 }
 
+point vehicle::pivot_point() const
+{
+    // TODO: a part type that let you override this
+    point p;
+    center_of_mass(p.x, p.y, false);
```

Comment:
Center of mass can be a relatively expensive operation in extreme cases (deathrollers), as it requires iterating over all items stored in the vehicle.
Later on (probably not in this PR) it may be worth caching it.